From 8311b9849420fc3894de1b08f393af8f26b60322 Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Wed, 19 Feb 2025 18:09:47 +0100
Subject: [PATCH 1545/1695] dts: stm32: add fixed IO compensation on vddio
 regulators for stm32mp211

IO compensation could alter duty cycle of high frequency output signal
on stm32mp21 SoC. To avoid such issue, fixed IO compensation is used.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5b939707acd0ab58d07df31c20ad87472af54946 dts: stm32: add fixed IO compensation on vddio regulators for stm32mp211]
Change-Id: Ibec16ed3ca21687ed218ddcef36cefaa180347e1
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/438580
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 4e04eb96d..51708cf4d 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -777,24 +777,28 @@
 			vddio1: vddio1 {
 				compatible = "st,stm32mp21-pwr-regu";
 				regulator-name = "vddio1";
+				st,iocomp = <8 7>;
 				status = "disabled";
 			};
 
 			vddio2: vddio2 {
 				compatible = "st,stm32mp21-pwr-regu";
 				regulator-name = "vddio2";
+				st,iocomp = <8 7>;
 				status = "disabled";
 			};
 
 			vddio3: vddio3 {
 				compatible = "st,stm32mp21-pwr-regu";
 				regulator-name = "vddio3";
+				st,iocomp = <8 7>;
 				status = "disabled";
 			};
 
 			vddio: vddio {
 				compatible = "st,stm32mp21-pwr-regu";
 				regulator-name = "vddio";
+				st,iocomp = <8 7>;
 				regulator-always-on;
 				status = "disabled";
 			};
-- 
2.47.2

